### PR TITLE
akira-unstable: init at 2019-10-12

### PR DESCRIPTION
**Pull request by Br1ght0ne**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---
This change is [Reviewable](https://reviewable.io/reviews/nixos/nixpkgs/74332)

### CODE REVIEW

**Comment by jonringer** [comment]
cc @worldofpeace for qt/gtk wrapping review :)

**Review by jonringer — APPROVED** [review]
`nix-review` passes on NixOS
diff LGTM
gui opens fine

```
[3 built, 3 copied (15.4 MiB), 2.3 MiB DL]
https://github.com/NixOS/nixpkgs/pull/74332
1 package were built:
akira
```

**Comment by jonringer** [comment]
@GrahamcOfBorg build akira

File: pkgs/applications/graphics/akira/default.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    ${glib.dev}/bin/glib-compile-schemas --strict $out/share/glib-2.0/schemas
+    ${gtk3.out}/bin/gtk-update-icon-cache -qtf $out/share/icons/hicolor
+    ${desktop-file-utils.out}/bin/update-desktop-database $out/share/applications
+  '';
+
+  meta = with lib; {
+    description = "Native Linux Design application built in Vala and GTK";
+    homepage = "https://github.com/akiraux/Akira";
+    license = licenses.gpl3;
+    maintainers = with maintainers; [ filalex77 ];
```
Comment (worldofpeace) [inline pkgs/applications/graphics/akira/default.nix]: This app is a elementary application :sparkles: Can you add `pantheon.maintainers`. (it's currently just me).

File: pkgs/applications/graphics/akira/meson-post-install.patch [hunk 2]
```diff
@@ -0,0 +1,10 @@
+diff --git a/meson.build b/meson.build
+index 9a4ad56..ce6c003 100644
+--- a/meson.build
++++ b/meson.build
+@@ -65,5 +65,3 @@ add_global_arguments(['-DG_LOG_DOMAIN="Akira"'], language:'c')
+ subdir('src')
+ subdir('data')
+ subdir('po')
+-
+-meson.add_install_script('build-aux/meson/post_install.py')
```
Comment (worldofpeace) [inline pkgs/applications/graphics/akira/meson-post-install.patch]: Please don't do this. What you probably need to do in the expression is add
`desktop-file-utils` and `python3` to `nativeBuildInputs` and patch the shebang https://github.com/NixOS/nixpkgs/blob/4e8c2f08b01486f2de0c7eb2468932e94b28030c/pkgs/desktops/pantheon/apps/elementary-code/default.nix#L86.

File: pkgs/applications/graphics/akira/default.nix [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    owner = "akiraux";
+    repo = "Akira";
+    rev = "cab952dee4591b6bde34d670c1f853f5a3ff6b19";
+    sha256 = "1fp3a79hkh6xwwqqdrx4zqq2zhsm236c6fhhl5f2nmi108yxz04q";
+  };
+
+  nativeBuildInputs = [ desktop-file-utils meson ninja pkgconfig vala_0_40 wrapGAppsHook ];
+  buildInputs = [
+    appstream-glib cairo glib goocanvas2 gtk3 gtksourceview3
+    json-glib libarchive libgee libxml2 meson pantheon.granite python3
```
Comment (worldofpeace) [inline pkgs/applications/graphics/akira/default.nix]: I don't think meson is going to be a runtime dependency here. No other references to python3 too
* https://github.com/akiraux/Akira/search?q=python3&unscoped_q=python3

File: pkgs/applications/graphics/akira/default.nix [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  version = "2019-10-12";
+
+  src = fetchFromGitHub {
+    owner = "akiraux";
+    repo = "Akira";
+    rev = "cab952dee4591b6bde34d670c1f853f5a3ff6b19";
+    sha256 = "1fp3a79hkh6xwwqqdrx4zqq2zhsm236c6fhhl5f2nmi108yxz04q";
+  };
+
+  nativeBuildInputs = [ desktop-file-utils meson ninja pkgconfig vala_0_40 wrapGAppsHook ];
```
Comment (worldofpeace) [inline pkgs/applications/graphics/akira/default.nix]: please use the latest vala `vala` attribute.

File: pkgs/applications/graphics/akira/default.nix [hunk 5]
```diff
@@ -0,0 +1,40 @@
+{ stdenv, lib, fetchFromGitHub
+, desktop-file-utils, meson, ninja, pkgconfig, vala_0_40, wrapGAppsHook
+, appstream-glib, cairo, glib, goocanvas2, gtk3, gtksourceview3
+, json-glib, libarchive, libgee, libxml2, pantheon, python3 }:
+
+stdenv.mkDerivation rec {
+  pname = "akira";
```
Comment (worldofpeace) [inline pkgs/applications/graphics/akira/default.nix]:
```suggestion
  pname = "akira-unstable";
```

File: pkgs/applications/graphics/akira/default.nix (hunk 3 above)
Comment (worldofpeace) [reply pkgs/applications/graphics/akira/default.nix]: `appstream-glib` is just used for appstream validation. it's native.

File: pkgs/applications/graphics/akira/default.nix (hunk 4 above)
Comment (worldofpeace) [reply pkgs/applications/graphics/akira/default.nix]: can you also add `vala-lint`  https://github.com/akiraux/Akira/blob/5f4fd22da5c06e1cefaa4859bffeb946874ca117/meson.build#L41 ?

File: pkgs/applications/graphics/akira/default.nix [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+  mesonFlags = [ "-Dprofile=default" ];
+
+  patches = [ ./meson-post-install.patch ];
+
+  postInstall = ''
+    ${glib.dev}/bin/glib-compile-schemas --strict $out/share/glib-2.0/schemas
+    ${gtk3.out}/bin/gtk-update-icon-cache -qtf $out/share/icons/hicolor
+    ${desktop-file-utils.out}/bin/update-desktop-database $out/share/applications
+  '';
```
Comment (worldofpeace) [inline pkgs/applications/graphics/akira/default.nix]: won't be needed when using the post-install script.

**Review by worldofpeace — CHANGES_REQUESTED** [review]
Can you please follow the code style like this expression
* https://github.com/NixOS/nixpkgs/blob/4e8c2f08b01486f2de0c7eb2468932e94b28030c/pkgs/desktops/pantheon/apps/elementary-code/default.nix

I've left some changes, mostly to match a better dependency declaration.

File: pkgs/applications/graphics/akira/default.nix (hunk 4 above)
Comment (filalex77) [reply pkgs/applications/graphics/akira/default.nix]: Sadly, can't use the latest `vala`: https://github.com/akiraux/Akira/issues/221

Comment (worldofpeace) [reply pkgs/applications/graphics/akira/default.nix]: this patch could fix this https://patch-diff.githubusercontent.com/raw/akiraux/Akira/pull/157.patch

Comment (worldofpeace) [reply pkgs/applications/graphics/akira/default.nix]: there's also https://github.com/akiraux/Akira/commit/bcda8fd53f6f232db0b6411269ba108af551629f.patch, this one is actually committed so maybe it's better. (not sure if it applies though).

File: pkgs/applications/graphics/akira/unstable.nix [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+, json-glib
+, libarchive
+, libgee
+, libxml2 }:
+
+stdenv.mkDerivation rec {
+  pname = "akira-unstable";
+  version = "2019-10-12";
+
+  repoName = "Akira";
```
Comment (worldofpeace) [inline pkgs/applications/graphics/akira/unstable.nix]: oh sorry, I actually didn't mean to suggest this. It just happened to be in that particularly expression.

Comment (filalex77) [reply pkgs/applications/graphics/akira/unstable.nix]: Sure, no worries. Was going to add the patch anyway :)

File: pkgs/applications/graphics/akira/default.nix (hunk 4 above)
Comment (filalex77) [reply pkgs/applications/graphics/akira/default.nix]: Sadly, the patch is not enough: https://gist.github.com/filalex77/31b9d5127616fbf710ec27d211626e62

Comment (filalex77) [reply pkgs/applications/graphics/akira/default.nix]: > there's also [akiraux/Akira@`bcda8fd`.patch](https://github.com/akiraux/Akira/commit/bcda8fd53f6f232db0b6411269ba108af551629f.patch), this one is actually committed so maybe it's better. (not sure if it applies though).

It applies! Thank you so much :smile:

**Comment by filalex77** [comment]
@GrahamcOfBorg build akira-unstable

File: pkgs/applications/graphics/akira/unstable.nix [hunk 8]
```diff
@@ -0,0 +1,57 @@
+{ stdenv, lib, fetchFromGitHub, appstream-glib, desktop-file-utils, meson, ninja
```
Comment (worldofpeace) [inline pkgs/applications/graphics/akira/unstable.nix]: one per line of the function formal arguments please.
```
{ stdenv,
, lib
, fetchFromGitHub
```

Comment (filalex77) [reply pkgs/applications/graphics/akira/unstable.nix]: Sorry, pesky formatter gets in the way when I don't notice. :eyes:

File: pkgs/applications/graphics/akira/default.nix (hunk 5 above)
Comment (jonringer) [reply pkgs/applications/graphics/akira/default.nix]: pname is used by repology to identify a given package. "akira-unstable" would be a different package entirely

Comment (jonringer) [reply pkgs/applications/graphics/akira/default.nix]: https://repology.org/project/akira/versions

Comment (jonringer) [reply pkgs/applications/graphics/akira/default.nix]: then again, it looks like none of the package repositories carry it

File: pkgs/top-level/all-packages.nix [hunk 9]
```diff
@@ -17839,6 +17839,8 @@ in
 
   airwave = callPackage ../applications/audio/airwave { };
 
+  akira-unstable = callPackage ../applications/graphics/akira/unstable.nix { };
```
Comment (jonringer) [inline pkgs/top-level/all-packages.nix]: you'll still want `akira/default.nix`. The unstable convention is used when designating that this was cut from master, not at an official release

**Comment by worldofpeace** [comment]
Oh sorry, it took me a bit to get back to this.

**Review by worldofpeace — APPROVED** [review]

**Comment by worldofpeace** [comment]
Thanks @filalex77 :sparkles:

**Pull request merged**